math/mathcore/inc/Fit/FitUtil.h

```diff
@@ -36,14 +36,14 @@
 #define USE_PARAMCACHE
 
 #ifdef R__HAS_VECCORE
-namespace vecCore{
+namespace vecCore {
 template <class T>
 vecCore::Mask_v<T> Int2Mask(unsigned i)
 {
    T x;
-   for (unsigned j = 0; j < vecCore::VectorSize<T>(); j++) vecCore::Set<T>(x, i, i + 1);
-   return vecCore::Mask_v<T>(x < i);
-   }
+   for (unsigned j = 0; j < vecCore::VectorSize<T>(); j++) vecCore::Set<T>(x, j, j + 1);
+   return vecCore::Mask_v<T>(x <= i);
+}
```

Comment:
It's not casting, but constructing explicitly. It's there to ensure correctness.
Note that this function is equivalent to yours, despite the minor changes in implementation.
The calling code can stay the same. If you call `Int2Mask<Double_v>(4)`, then 4 elements will be true, since `{0, 1, 2, 3}` are all less than 4. In your case, you have `{1, 2, 3, 4} <= 4`, which works the same, but needs to add 1 to each element when creating the vector. Note also that I don't put the body of the `for` loop on the same line. I think it can be confusing, especially if you don't leave a blank line after it.